### DISCUSSION TITLE
How to generate a new dataset for grid2op?

### DISCUSSION THREAD

**Discussion by mengfeiwu-ai** [rte-france/Grid2Op#388] [Q&A]
Hello, i am trying to produce a new dataset which has a huge power grid scale, and i need this dataset accord with the standard of Grid2Op. I have already got a grid.json from pandapower, and it works. But the chronics data  is a big problem. I try to generate the data by 'Chronic2grid', while i find it's hard to generate the files of the path '/getting_started/example/input/generation/ case118_l2rpn_wcci ', could you tell me how to produce these files, like 'loads_charac.csv' or 'params_load.json' so on.Or i can get these files by pandapower or grid2op api directly?
<img width="792" alt="path" src="https://user-images.githubusercontent.com/120641811/207835013-ea49cd20-6eb1-46fc-9931-05c93f955fb5.png">

**Comment by BDonnot** [comment]
Hello,

Thanks for reaching out.

For now the most transparent environment we have to this regard is the `l2rpn_wcci_2022`.

It has also some configuration files that are used by the `chronix2grid` package to generate the data for the environment.

These configuration files are present to parameterize the generation process. For example the `load_charac.csv` describe for each load its properties such as its location (used for spatio temporal correlation : the closer two loads are, the more correlated they will be) as well as their "max" (in fact it's more kind of maximum, in the sense that some time, some load might be above this "max") etc.

It's similar for all the other files.

You can have a look at the `chronix2grid` doc for more information about them.

Let us know if you want any more help on the matter or if that's enough for now.

Best

Benjamin

**Reply by mengfeiwu-ai** [reply]
Thank you so much for replying on this question, Now i have already solved the problem of some 'json' files of that path, But it seemed that the information of 'prods_charac.csv' and 'loads_charac.csv' is the description of the generators or the loads, bucause i saw some information in csv files such as the 'names ' ,'cost' or the 'types' of the generator ,  Can i generate these files or generate these information from the grid.json? How do you define these information when you generate a dataset like "l2rpn-wcci-2022"? If this work is too complicated, maybe i need to evaluate this work and start over.

**Reply by BDonnot** [reply]
Hello,

To be more explicit grid.json details the grid model, the equations and everything. It is related to the equations etc. In this file you don't care were the power come from1 MW from solar or from nuclear or gaz is the same MW. 

The other files, needed for chronix2grid, are there to detail how the demand and generation will "behave". The time serie generated for a nuclear generator is not the same as the one for gaz or wind for example. 

The two information are completely independent. So no you don't have the type of generator in the json (you could add it if you wanted of course for other purposes) 

For our competitions we decided with "expert knowledge" what type of generator are located where. It is for you to decide that. It's an input to chronix2grid. 

For real grid you know this information. But for simulated one it should be somehow invented. 

Ps. As for the work needed... It depends on what you want. If you want a grid that really looks like a real one, then yes it's a lot of work. If you want something that resembles something that could be then it's much easier, pretty much anything works.

Pps also it depends on the level of renewable that you want. The more renewables you want the harder it will be.
